src/workbench/workbench.js

```diff
@@ -0,0 +1,210 @@
+        result.set(measurement.label, {
+          label: measurement.label,
+          unit: undefined,
+          value: [],
+          revision: [],
+          timestamp: [],
+        });
+      }
+
+      const series = result.get(measurement.label);
+      assert(series.label == measurement.label);
+
+      if (series.unit) {
+        assert(series.unit == measurement.unit);
+      } else {
+        series.unit = measurement.unit;
+      }
+
+      series.value.push(measurement.value);
+      series.revision.push(run.revision);
+      series.timestamp.push(run.timestamp);
+    }
+  }
+  return result.values();
+}
+
+// Plot time series using <https://apexcharts.com>.
+function plotSeries(seriesList, rootNode) {
+  for (const series of seriesList) {
+    let options = {
+      title: {
+        text: series.label,
+      },
+      chart: {
+        type: "bar",
+        height: "400px",
+      },
+      series: [{
+        name: series.label,
+        data: series.timestamp.map((t, i) => [t * 1000, series.value[i]]),
+      }],
+      xaxis: {
+        type: "datetime",
+      },
+    };
+
+    if (series.unit === "bytes") {
+      options = Object.assign(options, {
+        yaxis: {
+          labels: {
+            formatter: formatBytes,
+          },
+        },
+      });
+    }
+
+    if (series.unit === "ms") {
+      options = Object.assign(options, {
+        yaxis: {
+          labels: {
+            formatter: formatDuration,
+          },
+        },
+      });
+    }
+
+    let div = document.createElement("div");
+    rootNode.append(div);
+    var chart = new ApexCharts(div, options);
+    chart.render();
+  }
+}
+
+function formatBytes(bytes) {
+  if (bytes === 0) return "0 Bytes";
+
+  const k = 1024;
+  const sizes = [
+    "Bytes",
+    "KiB",
+    "MiB",
+    "GiB",
+    "TiB",
+    "PiB",
+    "EiB",
+    "ZiB",
+    "YiB",
+  ];
+
+  let i = 0;
+  while (i != sizes.length - 1 && Math.pow(k, i + 1) < bytes) {
+    i += 1;
+  }
+
+  return parseFloat((bytes / Math.pow(k, i)).toFixed(2)) + " " + sizes[i];
+}
+
+function formatDuration(durationInMilliseconds) {
+  const milliseconds = durationInMilliseconds % 1000;
+  const seconds = Math.floor((durationInMilliseconds / 1000) % 60);
+  const minutes = Math.floor((durationInMilliseconds / (1000 * 60)) % 60);
+  const hours = Math.floor((durationInMilliseconds / (1000 * 60 * 60)) % 24);
+  const days = Math.floor(durationInMilliseconds / (1000 * 60 * 60 * 24));
+  const parts = [];
+
+  if (days > 0) {
+    parts.push(`${days}d`);
+  }
+  if (hours > 0) {
+    parts.push(`${hours}h`);
+  }
+  if (minutes > 0) {
+    parts.push(`${minutes}m`);
+  }
+  if (seconds > 0 || parts.length === 0) {
+    parts.push(`${seconds}s`);
+  }
+  if (milliseconds > 0) {
+    parts.push(`${milliseconds}ms`);
+  }
+
+  return parts.join(" ");
+}
+
+// Returns the ISO week of the date.
+//
+// Source: https://weeknumber.com/how-to/javascript
+Date.prototype.getWeek = function () {
```

Comment:
Perhaps a standalone fn rather than extending the built in prototype?

Reply:
Yeah...